src/history/hs.c

```diff
@@ -141,9 +141,16 @@ int
 __wt_hs_create(WT_SESSION_IMPL *session, const char **cfg)
 {
     WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    const char *drop_cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_drop), "force=true", NULL};
 
     conn = S2C(session);
 
+    /* The LAS table may exist on upgrade. Discard it. */
+    WT_WITH_SCHEMA_LOCK(
+      session, ret = __wt_schema_drop(session, "file:WiredTigerLAS.wt", drop_cfg));
+    WT_RET(ret);
```

Comment:
This might be the cause of the failure of the read only tests. We should not attempt any modification of a database in read-only mode. So probably move these lines below the read-only check.

Reply:
Yes, that fixed it.Thank you!